generators/typescript/sdk/client-class-generator/src/GeneratedWebsocketClientClassImpl.ts

```diff
@@ -0,0 +1,533 @@
+        const reference = context.websocket.getReferenceToWebsocketSocketClass(this.channelId);
+        return reference.getTypeNode({
+            isForComment: true
+        });
+    }
+
+    public getReferenceToSocket(context: SdkContext): ts.Expression {
+        const reference = context.websocket.getReferenceToWebsocketSocketClass(this.channelId);
+        return reference.getExpression();
+    }
+
+    public getReferenceToOptions(): ts.Expression {
+        return ts.factory.createPropertyAccessExpression(
+            ts.factory.createThis(),
+            GeneratedWebsocketClientClassImpl.OPTIONS_PRIVATE_MEMBER
+        );
+    }
+
+    public getReferenceToArgs(): ts.Expression {
+        return ts.factory.createIdentifier(GeneratedWebsocketClientClassImpl.CONNECT_ARGS_PRIVATE_MEMBER);
+    }
+
+    private getReferenceToBaseUrl(context: SdkContext): ts.Expression {
+        return context.coreUtilities.fetcher.Supplier.get(
+            this.getReferenceToOption(GeneratedWebsocketClientClassImpl.BASE_URL_OPTION_PROPERTY_NAME)
+        );
+    }
+
+    private getReferenceToEnvironment(context: SdkContext): ts.Expression {
+        return context.coreUtilities.fetcher.Supplier.get(
+            this.getReferenceToOption(GeneratedWebsocketClientClassImpl.ENVIRONMENT_OPTION_PROPERTY_NAME)
+        );
+    }
+
+    private getOptionKeyForAuthHeader(header: HeaderAuthScheme): string {
+        return header.name.name.camelCase.unsafeName;
+    }
+
+    private getBearerAuthOptionKey(bearerAuthScheme: BearerAuthScheme): string {
+        return bearerAuthScheme.token.camelCase.safeName;
+    }
+
+    private getBasicAuthUsernameOptionKey(basicAuthScheme: BasicAuthScheme): string {
+        return basicAuthScheme.username.camelCase.safeName;
+    }
+
+    private getBasicAuthPasswordOptionKey(basicAuthScheme: BasicAuthScheme): string {
+        return basicAuthScheme.password.camelCase.safeName;
+    }
+
+    public getBaseUrl(channel: WebSocketChannel, context: SdkContext): ts.Expression {
+        const referenceToBaseUrl = this.getReferenceToBaseUrl(context);
+
+        const environment = this.getEnvironment(channel, context);
+
+        return ts.factory.createBinaryExpression(
+            referenceToBaseUrl,
+            ts.factory.createToken(ts.SyntaxKind.QuestionQuestionToken),
+            environment
+        );
+    }
+
+    public instantiate(args: { referenceToClient: ts.Expression; referenceToOptions: ts.Expression }): ts.Expression {
+        return ts.factory.createNewExpression(ts.factory.createIdentifier(this.serviceClassName), undefined, [
+            args.referenceToOptions
+        ]);
+    }
+
+    public accessFromRootClient(args: { referenceToRootClient: ts.Expression }): ts.Expression {
+        throw new Error("Not implemented");
+    }
+
+    public instantiateAsRoot(args: { context: SdkContext; npmPackage?: NpmPackage }): ts.Expression {
+        throw new Error("Not implemented");
+    }
```

Comment:
Do we still need these? Are they required to implement some interface defined elsewhere?